app/controllers/api/v1/login_controller.rb

```diff
@@ -0,0 +1,21 @@
+require_relative '../../../lib/json_web_token'
+
+class Api::V1::LoginController < ApiController
+  def auth
+    @user = User.find_by_email(params[:email])
+    if @user&.valid_password?(params[:password])
+      token = JsonWebToken.encode(user_id: @user.id)
+      time = Time.now + 24.hours.to_i
+      render json: { token:, exp: time.strftime('%m-%d-%Y %H:%M'),
+                     username: @user.name }, status: :ok
+    else
+      render json: { error: 'unauthorized' }, status: :unauthorized
+    end
+  end
+
+  private
+
+  def login_params
+    params.permit(:email, :password)
+  end
+end
```

Comment:
## [OPTIONAL]

- [ ] Nice work adding authentication to your API with `jwt` gem 👏 , however, you just have the API login functionalities, and it would be nice making also an API sign-up functionality.